### ISSUE TITLE
Nemo - opening folder from panel starts second instance.

### ISSUE THREAD

**Issue by jlynton** [linuxmint/nemo#1314]
Mint 18 cinammon.

When I move an icon on the desktop it becomes double (two of the same icon), this can, but does not always, happen with folders or files. The same happens if I move a file or folder from a folder to the desktop. The icon is doubled, it has one 'real' (active, clickable, selectable) icon, plus a 'ghost' icon that looks identical but cannot be manipulated.

Occasionally this can be fixed by selecting the desktop and pressing F5, otherwise I can only tidy up by logging out the in again.

This behaviour only occurs when dropping files onto the desktop itself, if I move a file or folder to the desktop folder in nemo, it appears correctly on the desktop.

I tried to delete ~/.config/nemo/desktop-metadata, but that did not help. I have also toggled the keep desktop icons aligned, but that did not help.

At least one other person has reported this behaviour, as can be seen at https://forums.linuxmint.com/viewtopic.php?f=208&t=229665&p=1217977#p1217977

After some experimentation following some helpful advice, I have found that if I use the Dropbox tray icon to open the Dropbox folder, or open a folder using the 'files' icon next to 'Menu' in the panel, or open one from 'Places' in the Menu itself, it starts a second instance of Nemo which causes the desktop icon problem. Killing the second Nemo solves it, until of course, I open a folder again using one of the described methods.

To summarise, opening a folder from anywhere in the cinnamon panel will start a second instance of Nemo.

**Comment by jlynton** [comment]
I have just noticed the same behaviour when clicking on "open folder" on a file in Transmission.

**Comment by mtwebster** [comment]
Do you have nautilus installed?  Is it running maybe? (`ps -A | grep nemo`)  It could be interfering with desktop handling.

**Comment by jlynton** [comment]
I do not have nautilus installed - ps -A | grep nemo returns    "12133 ?        00:00:27 nemo"
ps -A | grep nautilus returns nothing.

**Comment by jlynton** [comment]
Caja is not running, or installed.
nemo --quit returns:
Initializing nemo-dropbox 3.0.0
sys:1: PyGIWarning: Nemo was imported without specifying a version first. Use gi.require_version('Nemo', '3.0') before import to ensure that the right version gets loaded.
Initializing folder-color-switcher extension...

If I kill nemo from the system monitor, I lose all the desktop icons.

**Comment by feren** [comment]
@jlynton Normally `nemo --quit` just returns nothing and closes all processes of Nemo running as the user with 'killall'...

**Comment by jlynton** [comment]
in my case it returns what I said previously, but does not kill nemo. I tried it at the suggestion of mtwebster 3 posts above.

**Comment by Odyseus** [comment]
Hello, everybody.

I couldn't reproduce the issue described by @jlynton if I opened several instances of Nemo from the menu or the panel launcher. There were always just one instance of the nemo process at all times. And then, I opened an instance of Nemo from Dropbox context menu, and Nemo went nuts. Using `nemo --quit` wouldn't kill the instance opened by Dropbox. Using `nemo --quit` a second time would throw an error similar to jlynton's.

```bash
$ nemo --quit
sys:1: PyGIWarning: Nemo was imported without specifying a version first. Use gi.require_version('Nemo', '3.0') before import to ensure that the right version gets loaded.
sys:1: Warning: /build/glib2.0-7IO_Yw/glib2.0-2.48.1/./gobject/gsignal.c:2635: instance '0x173f1a0' has no handler with id '315'
```

From this point, when opening Nemo from the menu or panel launcher will always start a second instance of the nemo process. Everything goes back to normal when I use `killall -9 nemo` and then open Nemo from menu or panel launcher, using always just one nemo process. The problem starts again if I open another instance of Nemo from Dropbox context menu.

I don't have installed Caja nor Nautilus nor any other software that could handle the desktop (that I know of). I don't use Nemo as my file manager, I use SpaceFM that, as far as I know, doesn't try to handle the desktop.

**Comment by mtwebster** [comment]
There's got to be something missing here.. I can't reproduce this in any way that's been mentioned.  It doesn't even make sense.

If you see that message, that's nemo starting up then quitting (it already wasn't running)

@Odyseus can you run through those steps again, up to where you do nemo --quit the first time (and there's still a window of it open?) - and run xprop in a terminal, then click that window.

After running nemo --quit, there *should be no process called nemo* running anymore - the only way I can see this happening is if one was started as root/sudo, but nemo shouldn't manage the desktop when it's running as root.

Can you guys tell me more details about how your system is setup - all running mint 18?  Multiple monitors?  Other logged in sessions running?

Hopefully I'm not coming across poorly - I'm truly baffled, and only trying to figure this out.

**Comment by Odyseus** [comment]
> can you run through those steps again, up to where you do nemo --quit the first time (and there's still a window of it open?) - and run xprop in a terminal, then click that window.

[This Gist](https://gist.github.com/Odyseus/b41abc6c53ff7ac269d55ef2b98fb689) contains the `xprop` output after clicking the Nemo window title bar opened by Dropbox that `nemo --quit` didn't kill.

> Can you guys tell me more details about how your system is setup - all running mint 18? Multiple monitors? Other logged in sessions running?

All my setups are single monitor. I always close a users session before logging into another user.

My setup using Mint 18 with Cinnamon 3.0.7 stable is mostly an standard/stable one. It only has one user account, never created a new user in this setup. The Gist that I linked above also contains the output of the [`inxi -F`](https://gist.github.com/Odyseus/b41abc6c53ff7ac269d55ef2b98fb689#file-nemoissue1314-002-txt) command (hardware info) and [`inxi -r`](https://gist.github.com/Odyseus/b41abc6c53ff7ac269d55ef2b98fb689#file-nemoissue1314-003-txt) command (repositories).

My setup using Mint 18 with Cinnamon nightly is almost like what I described above, but installed in a VirtualBox machine and it has two users created, but logged into just one at a time. Plus, the extra PPA for installing Cinnamon nightly. Here I can reproduce the exact behaviour that I described in my previous message, but as I started Dropbox from the terminal, it kept logging to terminal some errors when I activated the Dropbox context menu to open the Dropbox folder (I don't know if it has anything to do with the issue we are discussing here). This is the output:

```shell
(nemo:2781): Gtk-WARNING **: Failed to register client: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.gnome.SessionManager was not provided by any .service files
sys:1: PyGIWarning: Nemo was imported without specifying a version first. Use gi.require_version('Nemo', '3.0') before import to ensure that the right version gets loaded.
Nemo-Share-Message: Called "net usershare info" but it failed: La «red compartida» devolvió el error 255: mkdir failed on directory /var/run/samba/msg.lock: Permiso denegado
net usershare: cannot open usershare directory /var/lib/samba/usershares. Error No existe el archivo o el directorio
Please ask your system administrator to enable user sharing.
```

My setup using Arch Linux with Cinnamon 3.0.7, also installed on a VirtualBox machine, is infinitely more minimal and also mono-user. Here I can also reproduce the same behaviour that I described in my previous message and with a similar error message when opening Nemo from Dropbox context menu:

```shell
(nemo:1016): Gtk-WARNING **: Failed to register client: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.gnome.SessionManager was not provided by any .service files

(nemo:1016): Gtk-WARNING **: Allocating size to GtkGrid 0x7f214000d160 without calling gtk_widget_get_preferred_width/height(). How does the code know the size to allocate?
```

**Comment by mtwebster** [comment]
How are you starting Cinnamon?  Logging in via mdm or other DM,  or some other method?  org.gnome.SessionManager is provided by cinnamon-session, and should never be unavailable.

**Comment by Odyseus** [comment]
In my two Mint 18 systems I use auto-login. They both have in common that I changed the login theme to GTK, whose description is _"No theme, just pure GTK."_. In the Arch Linux system I use LightDM. I forgot to comment before, all of these systems are "pure Cinnamon". There are no other desktop environments installed.

Did you read the edit that I made to my previous message? Can you reproduce the issue by launching an application with `dbus-launch`? I just tried it launching `dbus-launch qbittorrent` and when I open a torrent folder from its interface, it throws the exact same error message and a second nemo process spawns. And if I open the Dropbox folder (Dropbox also launched with `dbus-launch`) from the Dropbox context menu, it starts a third nemo process.

I also performed some tests on Ubuntu Gnome. It happens exactly the same with nautilus on Gnome Shell and it throws the exact same message.

```shell
(nautilus:31508): Gtk-WARNING **: Failed to register client: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.gnome.SessionManager was not provided by any .service files
```

**Comment by mtwebster** [comment]
Ok, dbus-launch starts a new session bus (one without org.gnome.SessionManager), fails to see that nemo already exists, so instead of telling the existing nemo to just open a new window, it fires up a completely new and independent one.

I'm not going to mess with trying to change this behavior (it bothers the heck out of me but I don't think it's within my control,) so I'm just going to try to prevent nemo from attempting to manage the desktop if something else already is.

**Comment by lestcape** [comment]
@mtwebster, Probably what i say it's stupid, but just an idea.

Will be possible in nemo start, check always if there are another instance?

If this is the case, redirect the request for that existent instance and then kill the new duplicate instance in this case. Yes, could exist more than instance also for a moment, but just one will survive and you not going to mess with trying to change this behavior, but at less will not  survive several nemo instances.
